### DISCUSSION TITLE
color progress bars for percentage indication

### DISCUSSION THREAD

**Discussion by VPraharsha03** [fastfetch-cli/fastfetch#702] [Q&A] [answered]
Thanks a lot for this tool! I'm trying to find whether there is a way to display a colored progress bar for the usage info, something similar to what winfetch implemented 

![20240126_173101.jpg](https://github.com/fastfetch-cli/fastfetch/assets/56556436/ac87ef79-e528-49dd-903a-91387cb76ecf)

also how do i add custom ascii art ? i tried `--logo` by passing ascii.txt but it failed to load

**Comment by CarterLi** [comment, marked as answer]
1. You may try `--percent-type 2`. See `--help percent-type` for detail
2. It should work. Screenshot?

**Reply by VPraharsha03** [reply]
@CarterLi 

![image](https://github.com/fastfetch-cli/fastfetch/assets/56556436/10559169-7b2a-4ced-b9ff-dff08594d295)

I'd also tried `fastfetch --logo ~\.config\fastfetch\ascii.txt` as well but got the same error

Also how do I show up winget packages as well ? I tried this,
```json
"packages",
        {
            "winget": "true"
        },
```
Sorry as I'm new and finding it bit hard to properly understand the structure/schema

**Reply by CarterLi** [reply]
{
            "type": "packages",
            "winget": true
        },

**Reply by CarterLi** [reply]
> I'd also tried fastfetch --logo ~\.config\fastfetch\ascii.txt as well but got the same error

Try `--logo c:/users/yourname/.config/fastfetch/ascii.txt`

**Reply by VPraharsha03** [reply]
> 
>         {
>             "type": "packages",
>             "winget": true
>         },

still scoop & chocolatey are only being shown ; ascii is working

**Reply by CarterLi** [reply]
You may need to increase child process waiting timeout, because winget detection can be very slow.

See `fastfetch -h packages-winget`. Also you need to make sure that `winget list` works first ( not being blocked by prompting or something )

try

```json
{
  "general": {
    "processingTimeout": 5000
  }
}
```

**Reply by VPraharsha03** [reply]
hmm, that explains why winfetch used to be slow --

**Reply by VPraharsha03** [reply]
> > I'd also tried fastfetch --logo ~\.config\fastfetch\ascii.txt as well but got the same error
> 
> Try `--logo c:/users/yourname/.config/fastfetch/ascii.txt`

@CarterLi Hi, I recently updated to latest version of fastfetch and seems that default ascii color changed to purple , how do i make it match the info text color ? and i'm sure that my ascii txt doesn't have any color information either apart from ascii characters

**Reply by CarterLi** [reply]
Please upload your config file

**Reply by VPraharsha03** [reply]
@CarterLi so sorry github app didn't notify me of your reply..

I'm using this config https://github.com/VPraharsha03/dotfiles/blob/main/.config%2Ffastfetch%2Fconfig.jsonc

and this line to start fastfetch
os.execute('fastfetch --logo %USERPROFILE%/.config/fastfetch/ascii.txt')

**Reply by CarterLi** [reply]
Try `--logo-color-1 white`

**Reply by VPraharsha03** [reply]
curious to know, was any coloring scheme for info get changed in newer versions

**Reply by VPraharsha03** [reply]
> Try `--logo-color-1 white`

as an alternative i've used `$9` color placeholder in my logo to restore previous color

**Reply by VPraharsha03** [reply]
@CarterLi Is there a way to get the values or ANSI color codes of color placeholders ? I'm trying to change color of user@host to $9 (https://github.com/fastfetch-cli/fastfetch/wiki/Json-Schema#6123131-property-json-config--modules--modules-items--anyof--item-1--oneof--title--color--user)

**Reply by CarterLi** [reply]
https://github.com/fastfetch-cli/fastfetch/blob/dev/src/logo/builtin.c
